src/sketchup-stl/exporter.rb

```diff
@@ -130,14 +141,17 @@ def self.write_face_binary(file, scale, mesh, normal)
         polygons = mesh.polygons
         polygons.each do |polygon|
           if (polygon.length == 3)
-            norm = mesh.normal_at(polygon[0].abs)
-            file.write(norm.to_a.pack("e3"))
+            # e - Float: single-precision, little endian byte order
+            file.write(normal.to_a.pack("e3"))
             for j in vertex_order do
               pt = mesh.point_at(polygon[j].abs)
               pt = pt.to_a.map{|e| e * scale}
               file.write(pt.pack("e3"))
             end
-            file.write([0].pack("v"))
+            # 2-byte "Attribute byte count" spacer. Nonstandard use by some stl software
+            # to store color data. Was never widely supported. Should be 0. 
+            # C - Integer: 8-bit unsigned
+            file.write([0, 0].pack("CC"))
```

Comment:
I guess it depends on which spec we go by. Wikipedias says it's a single 16-bit unsigned int while fabbers says it's a 2 8-byte unsigned ints.  In practice there is no difference.  It was the "VAX" that threw me.  I think "S<" is what we want, although the table for `Array.pack` isn't completely clear. I'll try to find out. 

http://ruby-doc.org/core-2.0.0/Array.html#method-i-pack